src/storage/src/hummock/sstable/block.rs

```diff
@@ -31,32 +31,32 @@ pub const DEFAULT_ENTRY_SIZE: usize = 16;
 #[derive(Clone)]
 pub struct Block {
     /// Uncompressed entries data, with restart encoded restart points info.
-    data: Bytes,
+    data: Vec<u8>,
     /// Uncompressed entried data length.
     data_len: usize,
     /// Restart points.
     restart_points: Vec<u32>,
 }
 
 impl Block {
-    pub fn decode(buf: Bytes) -> HummockResult<Self> {
+    pub fn decode(buf: &[u8]) -> HummockResult<Self> {
         // Verify checksum.
         let xxhash64_checksum = (&buf[buf.len() - 8..]).get_u64_le();
         xxhash64_verify(&buf[..buf.len() - 8], xxhash64_checksum)?;
 
         // Decompress.
         let compression = CompressionAlgorithm::decode(&mut &buf[buf.len() - 9..buf.len() - 8])?;
         let compressed_data = &buf[..buf.len() - 9];
-        let buf = match compression {
-            CompressionAlgorithm::None => buf.slice(..buf.len() - 9),
+        let mut buf = match compression {
+            CompressionAlgorithm::None => buf[..buf.len() - 9].to_vec(),
```

Comment:
I think we can simply pass `Vec<u8>` as `buf` here to avoid a buffer copy if there is no compression. I've tried in #4259 and this part works.

FYI: https://github.com/singularity-data/risingwave/pull/4259/files#diff-118b861ab8b246f4481264ec796e38359c493bc9fc18e686cea8640d8d11da2fR38-R51

Reply:
And with `Vec<u8>`, we need to modify the return type of object store, to make it return `Vec<u8>`. This part is also zero cost compared with the current implementation.

Reply:
The only extra cost comes from `MultiBuilder::seal_current`, which requires both `Vec<u8>` and splited `Vec<Vec<u8>>`, since we cannot safely convert `Vec<u8>` into multiple continuous `Vec<Vec<u8>>`.

Reply:
We can resolve this problem in next PR.